distribution/lib/Standard/Base/0.2.32-SNAPSHOT/src/Data/Vector.enso

```diff
@@ -903,3 +900,21 @@ Singleton_Error.to_display_text : Text
 Singleton_Error.to_display_text =
     "The vector " + this.vec.to_text  + " has only one element."
 
+
+type Mixed_Types_Map_Key
+    type Mixed_Types_Map_Key value
+
+    == : Mixed_Types_Map_Key -> Boolean
+    == that = this.value == that.value
+
+    compare_to : Mixed_Types_Map_Key -> Ordering
+    compare_to that =
+        # Naive Compare
+        r = Panic.recover (this.value.compare_to that.value)
+        r.catch err->
+            bool_check = this.value.is_a Boolean && that.value.is_a Boolean
+            if bool_check then if (this.value then Ordering.Greater else Ordering.Less) else
+                this_type = Meta.get_qualified_type_name this.value
+                that_type = Meta.get_qualified_type_name that.value
+                if this_type != that_type then (this_type.compare_to that_type) else
+                    Panic.throw err
```

Comment:
These questions depend on what will go to replace `Panic.throw err`:

1. If we develop some way to compare the incomparable objects (like based on hashcode), I don't think differentiating their type gives us any advantage.
2. In the same vein, if we'll have a generic solution, making `Boolean` a special case also seems weird - why should it be treated specially? The only reason that comes to my mind is that we may want to enforce some specific ordering (like `False < True`) - but I think the whole reasoning behind Boolean not supporting `compare_to` is that we do not want to impose any ordering and the Map is really an unordered container (i.e. the order is rather an implementation detail), so I would avoid making this exception if possible.

Also, we may need to do the bucket-based solution if we want to handle the general case, as objects which do not support `compare_to` will need to be disambiguated at some point using O(n^2) `==` comparisons (although as discussed we may ensure that `n` is usually quite small by limiting comparisons to some small buckets, for example based on the hashcode).

Reply:
The `Boolean` special case was just a quick workaround until sorting the more general decision on supporting comparisions of booleans.

While I agree no obvious reason `True > False`, this is the convention in data space tools (R, Python, Excel, PostgreSQL) - solving that should be a separate ticket. I will talk with @NedHarding and @wdanilo to get agreement on whether to proceed for that one.

Reply:
Raised for further discussion